ibis/omniscidb/operations.py

```diff
@@ -313,6 +313,90 @@ def extract_field_formatter(translator, expr, sql_attr=sql_attr):
     return extract_field_formatter
 
 
+def _n_weeks_year(year: ir.NumericValue):
+    """
+    Return the number of the total of weeks for the given year.
+
+    Parameters
+    ----------
+    year : ibis.expr.types.NumericValue
+
+    Returns
+    -------
+    Ibis.expr.types.Expr
+
+    See Also
+    --------
+    https://en.wikipedia.org/wiki/ISO_week_date
+    """
+    # n_weeks_year adjustment
+    def _p(year):
+        return (year + (year // 4) - (year // 100) + (year // 400)) % 7
+
+    return 52 + (
+        ibis.case()
+        .when((_p(year) == 4) | (_p(year - 1) == 3), 1)
+        .else_(0)
+        .end()
+    )
+
+
+def _woy_preliminary(d: Union[ir.DateValue, ir.TimestampValue]) -> ir.Expr:
+    """
+    Return a preliminary week of year (WOY).
+
+    Parameters
+    ----------
+    d : ibis.expr.types.DateValue or ibis.expr.types.TimestampValue
+
+    Returns
+    -------
+    ir.Expr
+
+    See Also
+    --------
+    https://en.wikipedia.org/wiki/ISO_week_date
+    """
+    doy = d.day_of_year()
+    dow = d.day_of_week.index() + 1
+
+    result = doy - dow
+    result += 10
```

Comment:
I think it makes the code less readable to have all this, instead of simply:
```python
d.day_of_year() - d.day_of_week.index() + 11
```

Reply:
you're right ... I just follow the algorithm from https://en.wikipedia.org/wiki/ISO_week_date .. I forgot to refactor that. thanks!